axon-framework/events/event-handlers.md

```diff
@@ -1,26 +1,27 @@
-* Parameters of type `MetaData` will have the entire `MetaData` of an `EventMessage` injected.
-* Parameters annotated with `@Timestamp` and of type `java.time.Instant` \(or `java.time.temporal.Temporal`\) will resolve to the timestamp of the `EventMessage`. This is the time at which the event was generated.
-* Parameters annotated with `@SequenceNumber` and of type `java.lang.Long` or `long` will resolve to the `sequenceNumber` of a `DomainEventMessage`. This provides the order in which the event was generated \(within the scope of the aggregate it originated from\). It is important to note that `DomainEventMessage` **can only** originate from an Aggregate. Event which thus have been published directly on the `EventBus`/`EventGateway` are _not_ implementations of the `DomainEventMessage` and as such will not resolve a sequence number. 
-* Parameters assignable to message will have the entire `EventMessage` injected \(if the message is assignable to that parameter\). If the first parameter is of type message, it effectively matches an event of any type, even if generic parameters would suggest otherwise. Due to type erasure, Axon cannot detect what parameter is expected. In such case, it is best to declare a parameter of the payload type, followed by a parameter of type message.
-* Parameters of type `TrackingToken` will have the current `TrackingToken` related to the processed Event injected.
-* When using Spring and the Axon Configuration is activated \(either by including the Axon Spring Boot Starter module\), any other parameters will resolve to autowired beans, if exactly one injectable candidate is available in the application context. This allows you to inject resources directly into `@EventHandler` annotated methods.
+Arguably the most important parameter of an event handler is the first parameter which refers to the payload of an `EventMessage`.
+If the event handler does not need access to the payload of the message, you can specify the expected payload type on the `@EventHandler` annotation.
 
-You can configure additional `ParameterResolver`s by implementing the `ParameterResolverFactory` interface and creating a file named `/META-INF/service/org.axonframework.common.annotation.ParameterResolverFactory` containing the fully qualified name of the implementing class.
+Do not configure the payload type on the annotation if you want the payload to be passed as a parameter.
+For an exhaustive list of all parameters, we refer to [this](../messaging-concepts/supported-parameters-annotated-handlers.md#supported-parameters-for-event-handlers) section.
 
-In all circumstances, at most one event handler method is invoked per listener instance. Axon will search for the most specific method to invoke, using following rules:
+In all circumstances, at most one event handler method is invoked per listener instance.
```

Comment:
Do you mean "at least one event handler method is invoked per... "?

Reply:
No, it is intended as "at most." 

This is the specify that if an Event Handling Component contains, for example, three event handlers for `SomeEvent`, that only one of the handlers will be invoked.
So, the "listener" in this case refers to an Event Handling Component. Or differently put, the class containing the event handling functions.